### PR TITLE
Issue #3192: Better labels and help text for the Contact module settings form.

### PR DESCRIPTION
**Pull request by klonos**
https://github.com/backdrop/backdrop-issues/issues/3192

### CODE REVIEW

**Comment by herbdool** [comment]
I would flip the title and description of the first change. The title should be "Expose a ... checkbox..."

I don't think "threshold" is the correct word. It has almost the opposite meaning. From the dictionary: "the magnitude or intensity that must be exceeded for a certain reaction, phenomenon, result, or condition to occur or be manifested." Maybe just rename it "Limiting Emails".

**Comment by klonos** [comment]
>I would flip the title and description of the first change. The title should be "Expose a ... checkbox..."

@herbdool the point of this PR was to change that. As I explain in https://github.com/backdrop/backdrop-issues/issues/3192:

>About the "Show the 'Send yourself a copy' checkbox" label ...I cannot think of any other place in the admin UI where we offer an option to show/hide form elements. I think that we instead give things titles related to the functionality they enable/disable. ...

Fair enough about the other comment. I do not personally agree, but fair enough. I have posted it in the issue for further discussion.

File: core/modules/contact/contact.admin.inc [hunk 1]
```diff
@@ -24,33 +24,40 @@ function contact_settings_form($form, &$form_state) {
...
     '#type' => 'select',
-    '#title' => t('messages per'),
+    '#title' => t('Email send time window'),
+    '#title_display' => 'invisible',
     '#options' => array(0 => t('None (disabled)')) + backdrop_map_assoc(array(60, 180, 300, 600, 900, 1800, 2700, 3600, 10800, 21600, 32400, 43200, 86400), 'format_interval'),
     '#default_value' => $config->get('contact_threshold_window'),
+    '#prefix' => t('messages per'),
+  );
+  $form['threshold']['help'] = array(
+    '#type' => 'item',
```
Comment (opi) [inline core/modules/contact/contact.admin.inc:59]: What about using the `#type => 'help'` ? https://api.backdropcms.org/form_api#help

Comment (opi) [reply core/modules/contact/contact.admin.inc:59]: bad idea, the `help` item is not styled in seven. Forget this comment.

File: core/modules/contact/contact.admin.inc [hunk 2]
```diff
@@ -24,33 +24,40 @@ function contact_settings_form($form, &$form_state) {
   );
   $form['fields']['copy'] = array(
     '#type' => 'checkbox',
-    '#title' => t("Show the 'Send yourself a copy' checkbox"),
-    '#description' => t("Allow users to send themselves a copy on site-wide and personal contact forms."),
+    '#title' => t("Allow users to send themselves a copy of the submitted data."),
```
Comment (jenlampton) [inline core/modules/contact/contact.admin.inc]: Allow users to send themselves a copy of the submission.

File: core/modules/contact/contact.admin.inc [hunk 3]
```diff
@@ -24,33 +24,40 @@ function contact_settings_form($form, &$form_state) {
...
+    '#description' => t("Exposes a 'Send yourself a copy' checkbox on site-wide and personal contact forms."),
```
Comment (jenlampton) [inline core/modules/contact/contact.admin.inc]: Include a 'Send yourself a copy' checkbox on site-wide and personal contact forms

File: core/modules/contact/contact.admin.inc [hunk 4]
```diff
@@ -24,33 +24,40 @@ function contact_settings_form($form, &$form_state) {
...
     '#default_value' => contact_show_personal_copy_checkbox()
   );
 
   $form['threshold'] = array(
     '#type' => 'fieldset',
+    '#title' => t("Send limit"),
```
Comment (jenlampton) [inline core/modules/contact/contact.admin.inc:34]: Spam prevention

File: core/modules/contact/contact.admin.inc (hunk 1 above)
Comment (jenlampton) [reply core/modules/contact/contact.admin.inc]: Sounds like something else we should fix though!

**Review by jenlampton — COMMENTED** [review]
I've added a few recommendations for alternate text here.

File: core/modules/contact/contact.admin.inc (hunk 2 above)
Comment (jenlampton) [reply core/modules/contact/contact.admin.inc]: (since its not always data)

File: core/modules/contact/contact.admin.inc (hunk 4 above)
Comment (jenlampton) [reply core/modules/contact/contact.admin.inc:34]: I think we can use a more general heading here, and something that resonates more clearly with people.

**Comment by opi** [comment]
@jenlampton New PR https://github.com/backdrop/backdrop/pull/2302 with your comments addressed. I like the "Spam prevention" title.

**Comment by quicksketch** [comment]
Closing in favor of https://github.com/backdrop/backdrop/pull/2302

**Pull request closed without merging**